src/mem3/src/mem3_util.erl

```diff
@@ -101,6 +101,10 @@ write_db_doc(DbName, #doc{id=Id, body=Body} = Doc, ShouldMutate) ->
     {ok, #doc{body = Body}} ->
         % the doc is already in the desired state, we're done here
         ok;
+    {ok, #doc{body = Body1}} ->
```

Comment:
Previously this case was handled in `_`  clause on line `112/116` and we return `conflict` in such case. With your change we would be changing this semantic.

Reply:
Yeah... I should have added some more context to this change and the change @tonysun83 pointed out. The clause I added here is to allow updating of db_docs. Without the clause I added, this code path will never perform doc updates. Aside from using this code in the test case I added, the only codepath that uses `mem3_util:write_db_doc/3` is the core fabric db creation logic in `fabric_db_create.erl`.

I walked through the code and it seems like handling the case of updating a document is plenty reasonable given that we're using normal revision update mechanics where the document will only update if it's a proper descendant of the current value, and if that's the case, why would we ever want to not update the doc to the latest version?

Feedback welcome here and perhaps someone else has other thoughts on it, but the logic here seems appropriate enough to include that I added it even though the only immediate use case I'm aware of is for more test I added.

Reply:
Since we run this on creation logic only, I don't know if we'd ever be in a position to update the doc with a new body. It would either already be created, and we'd get a conflict, or it wouldn't be there, and we'd try to create it.


Even in general (if this wasn't just a fabric_db_create call), we might not know if `Body1` is necessarily an "updated" `Body`? It could be a stale value  too. Then, we would get a `conflict` thrown but we don't catch it here properly like we do in the `{not_found, _}` clause.

It may be safer to avoid modifying the subtle bits there especially with how conflict throws are handled?

Reply:
Well, I needed to update a dbs db doc, and this is the logical place to do it. My thoughts on the conflict issue are that `Body1` will by definition be an updated value if and only if the update succeeds, and if not then a conflict will be thrown, and I'm also not a fan of avoiding changing code to be safe from hypothetical errors.

I'm only using this from a test case, so it's not the end of the world if we take it out of this function, but between the options of: 1) duplicate the write functionality in the test suite, 2) add a new `mem3_util:update_db_doc` function, or 3) splice it in here, I went with option 3). If folks feel strongly, we could do option 2), but I would prefer to avoid option 1).

Reply:
I would opt to not altering the behavior just for the test. The conflict being thrown would also alter the overall logic in the surrounding code where we seem to catch the `conflict` throw and handle it as an error return as opposed to letting it propagate up the stack